### PR TITLE
Support register, unregister, and sideload of Office add-ins

### PR DESCRIPTION
**Pull request by akrantz**
Instead of using office-toolbox to register, unregister, and sideload, provide this capability in office-addin-dev-settings.

Change office-addin-debugging start/stop commands to use this integrated functionality. 

NOTE: this is a breaking change. 
* The startDebugging() function adds the "app" parameter to indicate the Office app, and removes the "sideload" parameter.
* The stopDebugging() function removes the "unload" parameter.

### CODE REVIEW

**Comment by akrantz** [comment]
@donaldgu @LouMM I'd appreciate it if you could please review.

File: packages/office-addin-debugging/src/commands.ts [hunk 1]
```diff
@@ -37,26 +39,21 @@ export async function start(manifestPath: string, appType: string | undefined, c
...
-        const sideload: string | undefined = command.sideload || process.env[`npm_package_scripts_sideload_${app}`] || process.env.npm_package_scripts_sideload;
 
         if (appTypeToDebug === undefined) {
             throw new Error("Please specify the application type to debug.");
         }
 
-        await startDebugging(manifestPath, appTypeToDebug, debuggingMethod, sourceBundleUrlComponents,
-            devServer, devServerPort, packager, packagerHost, packagerPort, sideload,
-            enableDebugging, enableLiveReload);
+        await startDebugging(manifestPath, appTypeToDebug, app, debuggingMethod, sourceBundleUrlComponents,
```
[the remark this answers is not in the document]
Comment (akrantz) [reply packages/office-addin-debugging/src/commands.ts]: Will consider for the future.

File: packages/office-addin-debugging/.vscode/launch.json [hunk 2]
```diff
@@ -27,7 +27,7 @@
       "type": "node",
       "request": "launch",
       "program": "${workspaceFolder}/lib/cli.js",
-      "args": [],
+      "args": ["start", "/Users/adamk/dev/git/Excel-Custom-Functions/manifest.xml"],
```
[the remark this answers is not in the document]
Comment (akrantz) [reply packages/office-addin-debugging/.vscode/launch.json]: Fixed.

File: packages/office-addin-dev-settings/src/dev-settings-mac.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+export async function registerAddIn(manifestPath: string, officeApps?: OfficeApp[]) {
+  try {
+    const manifest = await readManifestFile(manifestPath);
+
+    if (!officeApps) {
+      officeApps = getOfficeAppsForManifestHosts(manifest.hosts);
+
+      if (officeApps.length === 0) {
+        throw new Error("The manifest file doesn't specify any hosts for the Office add-in.");
```
[the remark this answers is not in the document]
Comment (akrantz) [reply packages/office-addin-dev-settings/src/dev-settings-mac.ts]: I've made it consistently "Office Add-in".

**Comment by akrantz** [comment]
"office-toolbox validate" needs to be supported as "office-addin-manifest validate".

**Comment by TCourtneyOwen** [comment]
Do these changes effectively make it so we can get rid off office-toolbox altogether? If they do and we do in fact plan to kill office-toolbox, then we will need to ensure that documentation is appropriately updated (e.g. tutorials)

**Comment by akrantz** [comment]
Yes, that's the goal. This pull request provides all of the functionality that from office-toolbox.

This pull request have the changes for Office-Addin-TaskPane to remove office-toolbox: https://github.com/OfficeDev/Office-Addin-TaskPane/pull/51. Note: it needs to be updated so the validate script uses `office-addin-manifest validate` instead of `office-toolbox validate`.

**Pull request merged**